include/fast_float/parse_number.h

```diff
@@ -60,6 +60,53 @@ from_chars_result parse_infnan(const char *first, const char *last, T &value)  n
   return answer;
 }
 
+/**
+ * Returns true if the floating-pointing rounding mode is to 'nearest'.
+ * It is the default on most system. This function is meant to be inexpensive.
+ * Credit : @mwalcott3
+ */
+fastfloat_really_inline bool rounds_to_nearest() noexcept {
+  // See
+  // A fast function to check your floating-point rounding mode
+  // https://lemire.me/blog/2022/11/16/a-fast-function-to-check-your-floating-point-rounding-mode/
+  //
+  // This function is meant to be equivalent to :
+  // prior: #include <cfenv>
+  //  return fegetround() == FE_TONEAREST;
+  // However, it is expected to be much faster than the fegetround()
+  // function call.
+  //
+  // The volatile keywoard prevents the compiler from computing the function
+  // at compile-time.
+  // There might be other ways to prevent compile-time optimizations (e.g., asm).
+  // The value does not need to be std::numeric_limits<float>::min(), any small
+  // value so that 1 + x should round to 1 would do.
+  static volatile float fmin = std::numeric_limits<float>::min();
+  //
+  // Explanation:
+  // Only when fegetround() == FE_TONEAREST do we have that
+  // fmin + 1.0f == 1.0f - fmin.
+  //
+  // FE_UPWARD:
+  //  fmin + 1.0f = 0x1.00001 (1.00001)
+  //  1.0f - fmin = 0x1 (1)
+  //
+  // FE_DOWNWARD or  FE_TOWARDZERO:
+  //  fmin + 1.0f = 0x1 (1)
+  //  1.0f - fmin = 0x0.999999 (0.999999)
+  //
+  //  fmin + 1.0f = 0x1 (1)
+  //  1.0f - fmin = 0x0.999999 (0.999999)
```

Comment:
float is on most CPUs IEEE single format, so in the fmin + 1.0f FE_UPWARD case it would be better
to mention:
0x1.000002
rather than
0x1.00001
because that is what one really gets in that format after that addition.
The decimal representation of that is
1.00000011920928955078125 if you want to mention it.
And 0x0.999999 is certainly wrong, the right value is
0x1.fffffe
hexadecimal and
0.999999940395355224609375
decimal.

Reply:
Let us take this stuff out. It is too long.

Note that my latest commit adds an optimization.

Reply:
I simplified the explanation in the latest commit.